tests/simple/testdata/optionals.textproto

```diff
@@ -234,4 +234,113 @@ section: {
       }
     }
   }
+  test {
+    name: "optional_empty_map_optindex_hasValue"
+    expr: "optional.of({}).?c.hasValue()"
+    value: {
+      bool_value: false
+    }
+  }
+  test {
+    name: "empty_struct_optindex_hasValue"
+    container: "google.api.expr.test.v1.proto2"
+    expr: "TestAllTypes{}.?repeated_string.hasValue()"
+    value: {
+      bool_value: false
+    }
+  }
+  test {
+    name: "optional_empty_struct_optindex_hasValue"
+    container: "google.api.expr.test.v1.proto2"
+    expr: "optional.of(TestAllTypes{}).?repeated_string.hasValue()"
+    value: {
+      bool_value: false
+    }
+  }
+  test {
+    name: "optional_none_optselect_hasValue"
+    expr: "optional.none().?repeated_string.hasValue()"
```

Comment:
This works? I would think the type-checker would complain about this

Reply:
Yeah, `optional.none()` should be `optional_type<dyn>`.